flink-table-store-core/src/main/java/org/apache/flink/table/store/file/stats/OrcFileStatsExtractor.java

```diff
@@ -0,0 +1,183 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.file.stats;
+
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.table.data.DecimalData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.data.TimestampData;
+import org.apache.flink.table.types.logical.DecimalType;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.table.utils.DateTimeUtils;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.orc.BooleanColumnStatistics;
+import org.apache.orc.ColumnStatistics;
+import org.apache.orc.DateColumnStatistics;
+import org.apache.orc.DecimalColumnStatistics;
+import org.apache.orc.DoubleColumnStatistics;
+import org.apache.orc.IntegerColumnStatistics;
+import org.apache.orc.OrcFile;
+import org.apache.orc.Reader;
+import org.apache.orc.StringColumnStatistics;
+import org.apache.orc.TimestampColumnStatistics;
+import org.apache.orc.TypeDescription;
+
+import java.io.IOException;
+import java.sql.Date;
+import java.util.List;
+import java.util.stream.IntStream;
+
+/** {@link FileStatsExtractor} for orc files. */
+public class OrcFileStatsExtractor implements FileStatsExtractor {
+
+    private final RowType rowType;
+
+    public OrcFileStatsExtractor(RowType rowType) {
+        this.rowType = rowType;
+    }
+
+    @Override
+    public FieldStats[] extract(Path path) throws IOException {
+        org.apache.hadoop.fs.Path hadoopPath = new org.apache.hadoop.fs.Path(path.toUri());
+        Reader reader =
+                OrcFile.createReader(hadoopPath, OrcFile.readerOptions(new Configuration()));
+
+        long rowCount = reader.getNumberOfRows();
+        ColumnStatistics[] columnStatistics = reader.getStatistics();
+        TypeDescription schema = reader.getSchema();
+        List<String> columnNames = schema.getFieldNames();
+        List<TypeDescription> columnTypes = schema.getChildren();
+
+        return IntStream.range(0, rowType.getFieldCount())
+                .mapToObj(
+                        i -> {
+                            RowType.RowField field = rowType.getFields().get(i);
+                            int fieldIdx = columnNames.indexOf(field.getName());
+                            int colId = columnTypes.get(fieldIdx).getId();
+                            return toFieldStats(field, columnStatistics[colId], rowCount);
+                        })
+                .toArray(FieldStats[]::new);
+    }
+
+    private FieldStats toFieldStats(RowType.RowField field, ColumnStatistics stats, long rowCount) {
+        long nullCount = rowCount - stats.getNumberOfValues();
```

Comment:
``` 
 /**
   * Get the number of values in this column. It will differ from the number
   * of rows because of NULL values and repeated values.
   * @return the number of values
   */
   long getNumberOfValues();
```
`getNumberOfValues` seems not work for null count.

Can we just store `hasNull` instead of `nullCount`?

Reply:
We can check nullCount with `hasNull`.

Reply:
I don't know why java docs of `ColumnStatistics#getNumberOfValues` state like this. But according to the implementation in `TreeWriterBase#writeBatch` and according to the unit tests current implementation is alright.